src/python/crash_analysis/stack_parsing/stack_analyzer.py

```diff
@@ -1199,789 +48,56 @@ def get_crash_data(crash_data,
-    if JNI_ERROR_STRING in state.crash_state and '/art/runtime/' in line:
-      continue
-
-    # Platform specific: Windows cdb style stack frame.
-    if add_frame_on_match(
-        WINDOWS_CDB_STACK_FRAME_REGEX,
-        line,
-        state,
-        group=4,
-        frame_spec=WINDOWS_CDB_STACK_FRAME_SPEC):
-      continue
-
-    # Platform specific: Linux and mac gdb, ASAN, MSAN, UBSAN style
-    # stack frame. Try the regex with symbols first i.e. with
-    # addresses and function names.
-    if add_frame_on_match(
-        SAN_STACK_FRAME_REGEX,
-        line,
-        state,
-        group=3,
-        frame_spec=SAN_STACK_FRAME_SPEC,
-        frame_override_func=llvm_test_one_input_override):
-      continue
-
-    # Chrome symbolized stack frame regex.
-    if add_frame_on_match(
-        CHROME_STACK_FRAME_REGEX,
-        line,
-        state,
-        group=4,
-        frame_spec=CHROME_STACK_FRAME_SPEC):
-      continue
-
-    # Chrome symbolized stack frame regex (Mac only).
-    if add_frame_on_match(
-        CHROME_MAC_STACK_FRAME_REGEX,
-        line,
-        state,
-        group=6,
-        demangle=True,
-        frame_spec=CHROME_MAC_STACK_FRAME_SPEC):
-      continue
-
-    # Chrome symbolized stack frame regex (Windows only).
-    if add_frame_on_match(
-        CHROME_WIN_STACK_FRAME_REGEX,
-        line,
-        state,
-        group=1,
-        frame_spec=CHROME_WIN_STACK_FRAME_SPEC):
-      continue
-
-    # Android java exception stack frames.
-    if (state.found_java_exception and
-        state.crash_type in ['CHECK failure', 'Fatal Exception'] and
-        add_frame_on_match(
-            JAVA_EXCEPTION_CRASH_STATE_REGEX, line, state, group=1)):
-      continue
-
-    # Android kernel stack frame.
-    android_kernel_match = add_frame_on_match(
-        ANDROID_KERNEL_STACK_FRAME_REGEX, line, state, group=3)
-    if android_kernel_match:
-      found_android_kernel_crash = True
-
-      # Update address from the first stack frame unless we already have
-      # more detailed information from KASan.
-      if state.frame_count == 1 and not is_kasan:
-        state.crash_address = '0x%s' % android_kernel_match.group(1)
-      continue
-
-    # V8 correctness fuzzer metadata.
-    if add_frame_on_match(
-        V8_CORRECTNESS_METADATA_REGEX,
-        line,
-        state,
-        group=1,
-        frame_filter=lambda s: s):
-      continue
-
-    # Golang stack frames.
-    if is_golang and add_frame_on_match(
-        GOLANG_STACK_FRAME_FUNCTION_REGEX,
-        line,
-        state,
-        group=1,
-        frame_filter=lambda s: s.split('/')[-1]):
-      continue
-
-    # Python stack frames.
-    if is_python and add_frame_on_match(
-        PYTHON_STACK_FRAME_FUNCTION_REGEX, line, state, group=3):
-      continue
-
-  # Only get repo.prop if we have an Android kernel or KASAN crash
-  if environment.is_android() and (found_android_kernel_crash or is_kasan):
+  # Linkify Android stacktrace.
```

Comment:
nit: maybe move this block into its own function. (this file is fine since hacky anyway).

Reply:
Done!